### PR TITLE
tests: improve test directory structure

### PR DESCRIPTION
**Pull request by squat**
I am updating the hierarchy of the Tectonic testing directory. The
intention of each directory is documented in `tests/README.md`. Future
commits will move the so-called "sanity" tests to the smoke test
directory and add integration tests to the corresponding location.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Tectonic Tests
+
+This directory contains all smoke and integration tests for Tectonic. Each subdirectory is further partitioned into directories for particular cloud providers as necessary. This directory should conform to the following directory hierarchy:
```
Comment (sym3tri) [inline tests/README.md]: s/cloud providers/platforms/

Not all are "cloud"

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+.
+├── gui           # Integration tests for the user interface
+├── integration   # Integration tests for Tectonic
+└── smoke         # Smoke tests for all platforms
+    ├── aws       # Smoke tests for AWS
+    │   └── vars  # Terraform tfvars files for AWS smoke tests
+    ├── azure     # Smoke tests for Azure
+    │   └── vars  # Terraform tfvars files for Azure smoke tests
+    └── baremetal # Smoke tests for baremetal
+        └── vars  # Terraform tfvars files for baremetal smoke tests
```
Comment (sym3tri) [inline tests/README.md]: Consider adding a `...` to indicate all future platforms should go here too

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Tectonic Tests
+
+This directory contains all smoke and integration tests for Tectonic. Each subdirectory is further partitioned into directories for particular cloud providers as necessary. This directory should conform to the following directory hierarchy:
+
+```
+.
+├── gui           # Integration tests for the user interface
```
Comment (sym3tri) [inline tests/README.md]: Maybe explicitly call this out as the "_Installer_ user interface" to avoid confusion with Console.

**Comment by sym3tri** [comment]
Those bash scripts are so small. Consider combining them all into a single file with sub-commands, invoked like: 

```sh
tests/smoke/aws/<script-name>.sh create
tests/smoke/aws/<script-name>.sh destroy
```

File: tests/smoke/aws/create.sh [hunk 4]
```diff
@@ -2,24 +2,10 @@
...
-# randomly select region
-REGIONS=(us-east-1 us-east-2 us-west-1 us-west-2)
-export CHANGE_ID=${CHANGE_ID:-${BUILD_ID}}
-i=$(( ${CHANGE_ID} % ${#REGIONS[@]} ))
-export TF_VAR_tectonic_aws_region="${REGIONS[$i]}"
-export AWS_REGION="${REGIONS[$i]}"
-echo "selected region: ${TF_VAR_tectonic_aws_region}"
+DIR="$( cd "$( dirname "$0" )" && pwd )"
+# shellcheck disable=SC1090
+source "$DIR/aws-common.sh"
```
Comment (Quentin-M) [inline tests/smoke/aws/create.sh]: where is this file?

File: tests/smoke/aws/create.sh [hunk 5]
```diff
@@ -2,24 +2,10 @@
...
-# Set the specified vars file
-TF_VARS_FILE=$1
-TEST_NAME=$(echo ${TF_VARS_FILE} | cut -d "." -f 1)
-
-# Set required configuration
-export PLATFORM=aws
-export CLUSTER="${TEST_NAME}-${BRANCH_NAME}-${BUILD_ID}"
-
-# s3 buckets require lowercase names
-export TF_VAR_tectonic_cluster_name=$(echo ${CLUSTER} | awk '{print tolower($0)}')
```
Comment (Quentin-M) [inline tests/smoke/aws/create.sh]: where did this logic go

Comment (squat) [inline tests/smoke/aws/create.sh]: this logic is in common.sh https://github.com/coreos/tectonic-installer/pull/930/files#diff-96cc1c7bce85cf8467a5311279f454ccR14

File: tests/README.md (hunk 1 above)
Comment (squat) [inline tests/README.md]: thanks, I switched between the two a couple of times. I'll revert to "platforms".

File: tests/smoke/aws/create.sh (hunk 5 above)
Comment (Quentin-M) [inline tests/smoke/aws/create.sh]: I see, thanks! I did not notice the weird rename `test/scripts/aws-destroy.sh → tests/smoke/aws/common.sh`.

**Comment by squat** [comment]
haha not often people say a shell script is too short! @sym3tri I condensed the scripts into a single `test.sh` at `tests/smoke/aws/test.sh` that exposes all the necessary commands: `create`, `destroy`, `plan`, and `test`, the last of which runs the actual smoke tests against a cluster.

**Comment by sym3tri** [comment]
@squat thanks, that's much more elegant IMO.

Small nit: invoking `test test` is a bit weird. Perhaps a better name for the script?

**Comment by squat** [comment]
@sym3tri yeah, I agree that's a bit funny. I would be in favor of renaming `test.sh -> smoke.sh`. This way the script name properly indicates its intended use.

**Comment by sym3tri** [comment]
Or `smoke test` :)

Either way. This could bikeshed forever. I'll leave it up to you.

**Comment by squat** [comment]
great I think we are in agreement. I'll make the change now. PTAL

**Comment by squat** [comment]
@sym3tri @ggreer PTAL. I'd like to merge so we can iterate on this and add more tests

**Comment by kans** [comment]
<3

**Pull request merged**